firmware/port.c

```diff
@@ -12,25 +12,32 @@ typedef enum PortCmd {
     CMD_NOP = 0,
     CMD_FLUSH = 1,
     CMD_ECHO = 2,
-    CMD_GPIO_IN = 3,
-    CMD_GPIO_HIGH = 4,
-    CMD_GPIO_LOW = 5,
-    CMD_GPIO_TOGGLE = 21,
-    CMD_GPIO_CFG = 6,
-    CMD_GPIO_WAIT = 7,
-    CMD_GPIO_INT = 8,
-
-    CMD_ENABLE_SPI = 10,
-    CMD_DISABLE_SPI = 11,
-    CMD_ENABLE_I2C = 12,
-    CMD_DISABLE_I2C = 13,
-    CMD_ENABLE_UART = 14,
-    CMD_DISABLE_UART = 15,
-    CMD_TX = 16,
-    CMD_RX = 17,
-    CMD_TXRX = 18,
-    CMD_START = 19,
-    CMD_STOP = 20,
+
+    CMD_GPIO_IN = 10, // switches pin to input mode AND reads value
+    CMD_GPIO_HIGH = 11, // switch to output and write high
+    CMD_GPIO_LOW = 12, // switch to output and write low
+    CMD_GPIO_TOGGLE = 13, // switch to output and toggle low/high
+    CMD_GPIO_CFG = 14, // ???
+    CMD_GPIO_WAIT = 15, // ???
+    CMD_GPIO_INT = 16, // set interrupt on pin
+    CMD_GPIO_INPUT = 17, // switches pin to input, does not read value
+    CMD_GPIO_OUTPUT = 18, // switches pin to output, does not write value
```

Comment:
Implementing it as `this.output(!!(Math.random() > 0.5))` would be about as sane.